include/nanobind/eigen/sparse.h

```diff
@@ -0,0 +1,177 @@
+            return false;
+        ScalarNDArray& values = data_caster.value;
+
+        if (object indices_o = obj.attr("indices"); !indices_caster.from_python(indices_o, flags, cleanup))
+            return false;
+        StorageIndexNDArray& inner_indices = indices_caster.value;
+
+        if (object indptr_o = obj.attr("indptr"); !indptr_caster.from_python(indptr_o, flags, cleanup))
+            return false;
+        StorageIndexNDArray& outer_indices = indptr_caster.value;
+
+        object shape_o = obj.attr("shape"), nnz_o = obj.attr("nnz");
+        Index rows, cols, nnz;
+        try {
+            if (len(shape_o) != 2)
+                return false;
+            rows = cast<Index>(shape_o[0]);
+            cols = cast<Index>(shape_o[1]);
+            nnz = cast<Index>(nnz_o);
+        } catch (const python_error &) {
+            return false;
+        }
+
+        value = SparseMap(rows, cols, nnz, outer_indices.data(), inner_indices.data(), values.data());
+
+        return true;
+    }
+
+    static handle from_cpp(T &&v, rv_policy policy, cleanup_list *cleanup) noexcept {
+        if (policy == rv_policy::automatic ||
+            policy == rv_policy::automatic_reference)
+            policy = rv_policy::move;
+
+        return from_cpp((const T &) v, policy, cleanup);
+    }
+
+    static handle from_cpp(const T &v, rv_policy policy, cleanup_list *) noexcept {
+        if (!v.isCompressed()) {
+            PyErr_SetString(PyExc_ValueError,
+                            "nanobind: unable to return an Eigen sparse matrix that is not in a compressed format. "
+                            "Please call `.makeCompressed()` before returning the value on the C++ end.");
+            return handle();
+        }
+
+        object matrix_type;
+        try {
+            matrix_type = module_::import_("scipy.sparse").attr(row_major ? "csr_matrix" : "csc_matrix");
+        } catch (python_error &e) {
+            e.restore();
+            return handle();
+        }
+
+        const Index rows = v.rows();
+        const Index cols = v.cols();
+        const size_t data_shape[] = { (size_t)v.nonZeros() };
+        const size_t outer_indices_shape[] = { (size_t)((row_major ? rows : cols) + 1) };
+
+        T *src = std::addressof(const_cast<T &>(v));
+        object owner;
+        if (policy == rv_policy::move) {
+            src = new T(std::move(v));
+            owner = capsule(src, [](void *p) noexcept { delete (T *) p; });
+        }
+
+        ScalarNDArray data(src->valuePtr(), 1, data_shape, owner);
+        StorageIndexNDArray outer_indices(src->outerIndexPtr(), 1, outer_indices_shape, owner);
+        StorageIndexNDArray inner_indices(src->innerIndexPtr(), 1, data_shape, owner);
+
+        try {
+            return matrix_type(make_tuple(
+                                   std::move(data), std::move(inner_indices), std::move(outer_indices)),
+                               make_tuple(rows, cols))
```

Comment:
I've noticed that this sometimes create ADL problems, because `nanobind::make_tuple` competes with `std::make_tuple`. A simple fix would be to fully qualify `make_tuple`, but is there a better solution? @wjakob 
```suggestion
            return matrix_type(nanobind::make_tuple(
                                   std::move(data), std::move(inner_indices), std::move(outer_indices)),
                               make_tuple(rows, cols))
```